src/components/apartment/SortApartments.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import useSessionStorage from '../../utils/useSessionStorage';
+
+const SortApartments = (items: any, sessionStorageID: string) => {
+  const sortDefaultProps = {
+    key: 'apartment_number',
+    direction: 'ascending',
+    alphaNumeric: true,
+  };
+
+  const [sortConfig, setSortConfig] = useSessionStorage({
+    defaultValue: sortDefaultProps,
+    key: `sortConfig-${sessionStorageID}`,
+  });
+
+  const sortedApartments = React.useMemo(() => {
+    if (items === undefined) return [];
+
+    let sortableApartments = [...items];
+
+    if (sortConfig !== null) {
```

Comment:
To save some indentation here, you could end early if `sortConfig === null` (or use the isEmpty mentioned earlier).
That way you can de-indent everything one step and just do `if (sortConfig === null) { return sortedApartments }.